osquery/tables/system/darwin/signature.mm

```diff
@@ -23,6 +22,34 @@
 namespace osquery {
 namespace tables {
 
+// Get the flags to pass to SecStaticCodeCheckValidityWithErrors, depending on
+// the OS version.
+Status getVerifyFlags(SecCSFlags& flags) {
```

Comment:
The original design was to start a worker poll and create distributed task queue style table selects. We had to pull back on that design because the SQLite virtual table API (and the way we use it) does not allow concurrent executions. This gets weird but essentially converges on the lack of ability to provide a "context" or "user data" arena into `sqlite_exec`. So once we lose execution into SQL and the subsequent calls to our table implementation module code we board a roller coaster of fun and hope for the best. 

Now! Your specific use of `SQL::selectAllFrom` is designed to bypass the SQL abstraction and call into the virtual table implementation directly. This **could** be threaded!

The "single query at a time" concept is actually pretty cool. Since this is an agent running on clients and production servers, you want to play as nice as possible with the CPU, filesystem, and memory. We do a LOT of work to assure utilization smoothing (splaying intervals, a nanny process that restricts CPU cycle/s and RSS, and bad-query blacklisting/backoffs). It also forces us to prefer a do-work-once-and-cache sort of thinking. :)